src/makeCalculator.js

```diff
@@ -37,7 +37,44 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  const calculator = {
+    add(x) {
+      result += x;
+    },
+
+    subtract(x) {
+      result -= x;
+    },
+
+    multiply(x) {
+      result *= x;
+    },
+
+    divide(x) {
+      result /= x;
+    },
+
+    reset() {
+      result = 0;
+
+      return calculator;
+    },
+
+    operate(callback, value) {
+      callback.call(this, value);
+
+      return calculator;
```

Comment:
```suggestion
      return this;
    },

    operate(callback, value) {
      callback.call(this, value);

      return this;
```

Reply:
it should work with any entity of this calculator

Reply:
I suggest you make an experiment: try to create two calculators and work with them. ask in the chat if you have questions